packages/web/components/funnels/concentrated-liquidity/swap-tool-promo.tsx

```diff
@@ -0,0 +1,107 @@
+import { AppCurrency } from "@keplr-wallet/types";
+import { ObservableQueryPool } from "@osmosis-labs/stores";
+import { useFlags } from "launchdarkly-react-client-sdk";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import { FunctionComponent, useMemo } from "react";
+import { useTranslation } from "react-multi-lang";
+
+import { Button } from "~/components/buttons";
+
+/** Show link to pools page in promo drawer if the send or out currency in swap tool is in the given list of pools.
+ *  Returns null if no pools are found containing the send or out currency.
+ */
+export const SwapToolPromo: FunctionComponent<{
+  pools: ObservableQueryPool[];
+  sendCurrency: AppCurrency;
+  outCurrency: AppCurrency;
+}> = ({ pools, sendCurrency, outCurrency }) => {
+  const router = useRouter();
+  const t = useTranslation();
+
+  // get the first denom to be found in CL pool
+  const denomInClPool = useMemo(() => {
+    const concentratedLiquidityPools = pools.filter(
```

Comment:
how big can the 'pools' array get? (just trying to get a sense of scale)

Reply:
There's at most 1000 pools, but these pools get filtered down to ~200-300 routable pools that we use for swapping. 80% of pools have nearly no liquidity

Reply:
That's fine then. If it was in the tens or hundreds of thousands then I think we might be able to make this function more efficient, but for 1300 that's def no problem